src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -83,7 +84,6 @@ public IServiceProvider Services
         {
             get
             {
-                EnsureApplicationServices();
```

Comment:
Does this means Services is null until Start is called?

Reply:
No. The WebHostBuilder always calls Initialize which sets the Services. I removed this Ensure line because it was redundant. 
https://github.com/aspnet/Hosting/blob/311a92ddee5a009b2e96beb18838e816b66d533b/src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs#L193

Reply:
Ah good